src/SAD/Data/Text/Block.hs

```diff
@@ -57,18 +58,19 @@ data Block = Block {
   declaredVariables :: Set Decl,
   name              :: Text,
   link              :: [Text],
-  position          :: SourcePos,
   tokens            :: [Token] }
   deriving (Eq, Ord)
 
-makeBlock :: Formula -> [ProofText] -> Section -> Text -> [Text] -> SourcePos -> [Token] -> (Block)
-makeBlock form body kind name link pos toks =
-  Block form body kind mempty name link (rangePos (tokensRange toks)) toks
+makeBlock :: Formula -> [ProofText] -> Section -> Text -> [Text] -> [Token] -> Block
+makeBlock form body kind = Block form body kind mempty
+
+position :: Block -> SourcePos
+position = rangePos . tokensRange . tokens
```

Comment:
Does that change the behavior somehow? In other words, did you check that all blocks are created (and updated) using `makeBlock`?